tsl/src/planner.c

```diff
@@ -190,13 +219,20 @@ tsl_create_distributed_insert_path(PlannerInfo *root, ModifyTablePath *mtpath, I
 
 	/* Check if it is possible to use COPY in the backend.
 	 *
-	 * There are two cases where we cannot use COPY:
+	 * There are three cases where we cannot use COPY:
 	 *
 	 * 1. ON CONFLICT clause exists
 	 *
 	 * 2. RETURNING clause exists and tuples are expected to be modified on
 	 *    INSERT by a trigger.
 	 *
+	 * 3. INSERT.. SELECT case. If the relation being inserted into and the
+	 *    relation in the SELECT are both distributed hypertables then do not
+	 *    use COPY. This is because we will need two connections to the same set
+	 *    of datanodes (one to ingest data into src table using COPY and one to
+	 *    SELECT from dst table) which is not supported currently in a single
+	 *    transaction. This restriction will be lifted in the future.
```

Comment:
Clarify that multiplexing multiple queries are normally possible on a single connection, but COPY is special as it switches the connection to the COPY_IN state.

```suggestion
	 * 3. INSERT.. SELECT case. If the relation being inserted into and the
	 *    relation in the SELECT are both distributed hypertables then do not
	 *    use COPY. This is because only one connection is maintained to data
	 *    nodes in order to maintain a single snapshot. Normally, it is possible to 
	 *    multiplex multiple queries on a connection using a CURSOR, but with
	 *    COPY the connection is switched into a COPY_IN state, making
	 *    multiplexing impossible. It might be possible to use two connections to 
	 *    the same data node (one to ingest data into src table using COPY and 
	 *    one to SELECT from dst table), but that requires coordinating on a single
	 *    snapshot to use on both connections. This is a potential future 
	 *    optimization.
```

Reply:
Ok, amended